portal-api/src/v1/api.js

```diff
@@ -471,6 +471,36 @@ const getPortalFacilityAmendment = async (facilityId, amendmentId) => {
   }
 };
 
+/**
+ * Upserts a draft amendment for a portal facility in the database.
+ * @param {Object} params
+ * @param {string} params.dealId - id of the deal with the relevant facility.
+ * @param {string} params.facilityId - id of the facility to amend.
+ * @param {import('@ukef/dtfs2-common').PortalFacilityAmendmentUserValues} params.amendmentUpdate - the draft amendment to be upserted.
+ * @param {import('@ukef/dtfs2-common').AuditDetails} params.auditDetails - The audit details for the update.
+ * @returns {Promise<(import('@ukef/dtfs2-common').PortalFacilityAmendmentWithUkefId)>} - the amendment
+ */
+const putPortalFacilityAmendment = async ({ dealId, facilityId, amendmentUpdate, auditDetails }) => {
+  try {
+    const response = await axios({
```

Comment:
Do we need to validate arguments since this is a JS file?

Reply:
`dealId` and `facilityId` are validated in a couple of places - before they they get to this layer (in Gef-UI we check they are valid Mongo IDs) but also in the portal-api middleware. `amendmentUpdate` and `auditDetails` are validated as part of middleware in central-api. 

So all of these parameters are validated already,  I think fine to not repeat that here.

Reply:
^ I've just added zod validation to the return type to make sure we do indeed get a `PortalFacilityAmendmentWithUkefId` from dtfs-central